python-sdk/pyproject.toml

```diff
@@ -82,7 +87,9 @@ all = [
     "sqlalchemy-bigquery>=1.3.0",
     "s3fs",
     "protobuf<=3.20", # Google bigquery client require protobuf <= 3.20.0. We can remove the limitation when this limitation is removed
-    "openlineage-airflow>=0.17.0"
+    "openlineage-airflow>=0.17.0",
+    "apache-airflow-providers-microsoft-azure",
+    "azure-storage-blob"
```

Comment:
I think we need to specify the minimum version for these two libraries

Reply:
@phanikumv do you have any recommendations for the minimum version?

We noticed that other providers, such as:
- apache-airflow-providers-amazon
- apache-airflow-providers-postgres
- apache-airflow-providers-snowflake
- apache-airflow-providers-databricks
are not pinned at the moment.
https://github.com/astronomer/astro-sdk/blob/main/python-sdk/pyproject.toml#L77-L89

And we are not sure what to do.

Reply:
@phanikumv we make the test with `apache-airflow-providers-microsoft-azure>=4.0.0`, but we have the following issue during the test.

- https://github.com/astronomer/astro-sdk/actions/runs/3716469536/jobs/6302839191#logs

```
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
nox > Session test-3.8(airflow='2.2.5') failed.
Error: Process completed with exit code 1.
```
We remove the version tag to keep testing okay. Let me know if we need to make more tests in this case.